Update README to mention releases and GitHub Action setup

blunderingsavant opened the pull request:
This PR addresses issue #78 by updating the README to provide better installation and setup instructions.

## Changes Made

### README.md Updates
- **Added GitHub Action setup as the recommended installation method** with complete step-by-step instructions including:
  - Prerequisites (bot account setup, Personal Access Token, Anthropic API)
  - Repository variables and secrets configuration
  - Complete workflow YAML with proper permissions and event triggers
- **Added pre-built binary installation option** with download links to the releases page and installation commands for different platforms
- **Added Go install option** for users who already have Go installed
- **Added Contributing section** with link to the new development guide

### New CONTRIBUTING.md
- **Moved development setup instructions** from README to a dedicated contributing guide
- **Enhanced development instructions** with detailed steps for:
  - Local environment setup with required tools (Just, golangci-lint)
  - Environment configuration for development
  - Running the bot locally in different modes
  - Testing and code quality checks
  - Submission process for contributions
- **Added project structure overview** to help new contributors understand the codebase
- **Included coding standards reference** pointing to the existing STYLE_GUIDE.md

## Why These Changes

1. **Improved User Experience**: The GitHub Action approach is much easier for end users compared to building from source
2. **Multiple Installation Options**: Users can choose the method that best fits their needs (GitHub Action for automation, binary for manual use, Go install for developers)
3. **Better Organization**: Development instructions are now properly separated from end-user instructions, following standard open-source project conventions
4. **Comprehensive Setup**: The GitHub Action instructions include all necessary configuration steps that were missing before

The changes maintain all existing functionality while making the project much more accessible to users who want to deploy the bot without setting up a full development environment.

Fixes #78

---
*This PR was created by the Blundering Savant bot.*

cchalm commented on README.md:
```diff
@@ -11,75 +11,184 @@ Generative AI is fallible in ways similar to people. It makes typos, misintepret
+
+#### Prerequisites
+
+1. **Create a Bot GitHub Account**:
```
Creating a GitHub account for the bot is required regardless of deployment mechanism, so let's move this to a section separate from the different deployment options.

Same for generating the github and anthropic tokens

cchalm commented on README.md:
```diff
@@ -11,75 +11,184 @@ Generative AI is fallible in ways similar to people. It makes typos, misintepret
+     - `ANTHROPIC_API_KEY`: Your Anthropic API key
+
+3. **Add the GitHub Action Workflow**:
+   - Create `.github/workflows/bot.yml` in your repository with the following content:
```
We could link to the workflow file within this repository, use that as the source of truth, what do you think?

cchalm commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,176 @@
+- Follow Go naming conventions (camelCase for private, PascalCase for public)
+- Write comprehensive tests for new functionality
+- Document exported functions and types
+- Handle errors explicitly
```
Don't duplicate content from the style guide, just link to it

blunderingsavant replied:
Agreed, I'll remove the duplicated content from the style guide and just provide a link to `STYLE_GUIDE.md` instead.

blunderingsavant commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,176 @@
+   - Any testing performed
+   - Screenshots or examples if applicable
+
+### Available Just Commands
```
Makes sense, I'll remove this redundant section since the help command is already mentioned above.

blunderingsavant commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,176 @@
+- `just logs` - View bot logs
+- `just clean` - Clean build artifacts
+
+### Project Structure
```
Good point about the codebase volatility. I'll remove the detailed project structure section to avoid it getting out of sync.

cchalm commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,150 @@
+1. **Go**: Install Go 1.24 or later from [https://golang.org/dl/](https://golang.org/dl/)
+2. **Git**: Ensure you have Git installed
+3. **Docker**: Install Docker for containerized development and testing
+4. **GitHub Account**: You'll need a GitHub account for testing and contributing
```
Remove the github account prerequisite, it's obvious

cchalm commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,150 @@
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/HEAD/install.sh | sh -s -- -b $(go env GOPATH)/bin
+```
+
+3. **Set up configuration**:
````
Remove this section and mention copying the environment template in `Environment Configuration` instead

cchalm commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,150 @@
+# - An Anthropic API key
+```
+
+4. **Build and test**:
````
Remove this section, as we have a more comprehensive section below called `Running the Bot Locally`

cchalm commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,150 @@
+
+This project follows the Go style guide outlined in [`STYLE_GUIDE.md`](STYLE_GUIDE.md). Please refer to that document for detailed coding standards and best practices.
+
+### Submitting Changes
```
This section describes the standard workflow for contributing to any open-source project. Since there is nothing unique to this project in this section, let's remove it

cchalm commented on README.md:
```diff
@@ -11,75 +11,108 @@ Generative AI is fallible in ways similar to people. It makes typos, misintepret
+   - The workflow file contains all the necessary configuration for running the bot automatically on issue assignments and PR updates
 
-### Configuration
+### Option 2: Pre-built Binary
```
Advantages: when run directly, the bot can handle tasks in multiple repositories. It also doesn't require any setup in the repositories to which the bot is contributing. Mention these advantages here

blunderingsavant commented:
I've addressed all the feedback from the latest review:

## Changes Made

### CONTRIBUTING.md Simplifications
- **Removed GitHub account prerequisite** - this was obvious and unnecessary
- **Consolidated environment setup** - merged the "Local Development Environment" section into "Environment Configuration" and moved the mention of copying `.env.example` there
- **Removed redundant "Running the Bot Locally" section** - there was already a more comprehensive section below covering the same content  
- **Removed standard open-source workflow** - the "Submitting Changes" section described generic processes that don't need to be documented for every project

### README.md Enhancement
- **Added binary advantages** - noted that direct binary usage supports multiple repositories and requires no repository setup

The documentation is now more focused and follows typical open-source project conventions without duplicating obvious information or standard practices. The development setup is streamlined while still providing all necessary information for contributors.
